ipaserver/plugins/certmapping.py

```diff
@@ -239,11 +248,186 @@ class cert_get_requestdata(Command):
+                for mapping in profile['ipacertfieldmapping']]
+        for mapping in field_mappings:
+            syntax_ruleset_name = mapping['ipacertsyntaxmapping'][0]
+            syntax_ruleset = api.Command.certmappingrule_show(syntax_ruleset_name['cn'])['result']
+            data_ruleset_names = mapping['ipacertdatamapping']
+            data_rulesets = [api.Command.certmappingrule_show(name['cn'])['result']
+                    for name in data_ruleset_names]
+
+            values = [self.evaluate_ruleset(ruleset, helper, **command_args) for ruleset in data_rulesets]
+            field_values.append(self.evaluate_ruleset(syntax_ruleset, helper, values=values, **command_args))
+
+        formatter = self.HELPERS[helper]()
+        formatted_values = formatter.format(field_values)
+        return formatted_values
+
+    def evaluate_ruleset(self, ruleset, helper, **kwargs):
+        use_rule = None
+        for rule_dn in ruleset['ipacerttransformation']:
+            rule = api.Command.certtransformationrule_show(rule_dn['cn'])['result']
+            if helper in rule['ipacerttransformationhelper']:
+                use_rule = rule
+                break
+        if use_rule is None:
+            raise errors.NotFound(
+                    reason=_('No transformation in "%(ruleset)s" rule supports format "%(helper)s"')
+                    % {'ruleset': ruleset['cn'][0], 'helper': helper})
+
+        template = rule['ipacerttransformationtemplate'][0]
+        if template.startswith('py:'):
+            template = template[3:]
+            template_method = getattr(self.Backend.datamapping, template)
+            return template_method(kwargs)
+        else:
+            raise NotImplementedError('Only py rules allowed for now')
+
+@register()
+class datamapping(Backend):
+    def _subject_base(self):
+        config = api.Command['config_show']()['result']
+        subject_base = config['ipacertificatesubjectbase'][0]
+        return subject_base
+
+    def syntaxSubjectOpenssl(self, extra_inputs):
+        section = 'distinguished_name = %%section(%s)' % extra_inputs['values'][0]
+        return section
+
+    def dataHostOpenssl(self, extra_inputs):
+        principal = extra_inputs['principal']
+        if 'subject' in principal:
+            subject = principal['subject']
+        else:
+            _, principal_name, _ = split_any_principal(principal['krbprincipalname'][0])
+            subject = 'CN=%s,%s' % (principal_name, self._subject_base())
+        return subject.replace(u',', u'\n')
+
+    def syntaxSubjectCertutil(self, extra_inputs):
+        arg = '-s %s' % pipes.quote(extra_inputs['values'][0])
+        return arg
+
+    def dataHostCertutil(self, extra_inputs):
+        principal = extra_inputs['principal']
+        if 'subject' in principal:
+            subject = principal['subject']
+        else:
+            _, principal_name, _ = split_any_principal(principal['krbprincipalname'][0])
```

Comment:
Done. Well, actually this code just disappeared because of the new way principals are handled.